_LowLevelCode/cpp/test/combine_sqw.tests/combine_sqw.test.cpp

```diff
@@ -10,820 +10,832 @@
-        ASSERT_EQ(pixels[pix_start_num * 9 + i], pix_buffer[start_buf_pos * 9 + i]);
-    }
+  start_buf_pos = 2;
+  reader.get_pix_for_bin(NUM_BINS_IN_FILE - 1, pPix_info, start_buf_pos,
+                         pix_start_num, num_bin_pix, false);
+  EXPECT_EQ(pix_start_num, sample_pix_pos[NUM_BINS_IN_FILE - 1]);
+  EXPECT_EQ(num_bin_pix, sample_npix[NUM_BINS_IN_FILE - 1]);
+  for (std::size_t i = 0; i < num_bin_pix * 9; i++) {
+    EXPECT_EQ(pixels[pix_start_num * 9 + i], pix_buffer[start_buf_pos * 9 + i]);
+  }
 }
 
-TEST_F(TestCombineSQW, DISABLED_SQW_Reader_Read_All) {
-    sqw_reader reader;
-    fileParameters file_par;
-    file_par.fileName = test_file_name;
-    file_par.file_id = 0;
-    file_par.nbin_start_pos = bin_pos_in_file;
-    file_par.pix_start_pos = pix_pos_in_file;
-    file_par.total_NfileBins = num_bin_in_file;
-    std::vector<float> pix_buffer;
-    pix_buffer.resize(this->pixels.size());
-    float *pPix_info = &pix_buffer[0];
-    size_t start_buf_pos(0), pix_start_num, num_bin_pix;
-    // --------------------------------------------------------------------------------------------
+TEST_F(TestCombineSQW, SQW_Reader_NoBuf_Mode) {
+  sqw_reader reader;
+  fileParameters file_par;
+  file_par.fileName = TEST_FILE_NAME;
+  file_par.file_id = 0;
+  file_par.nbin_start_pos = BIN_POS_IN_FILE;
+  file_par.pix_start_pos = PIX_POS_IN_FILE;
+  file_par.total_NfileBins = NUM_BINS_IN_FILE;
+  bool initialized(false);
+  try {
     reader.init(file_par, false, false, 0);
+    initialized = true;
+  } catch (...) {
+  }
 
-    auto t_start = std::chrono::steady_clock::now();
-    start_buf_pos = 0;
-    for (size_t i = 0; i < this->num_bin_in_file; i++) {
-        reader.get_pix_for_bin(i, pPix_info, start_buf_pos, pix_start_num, num_bin_pix, false);
-        start_buf_pos += num_bin_pix;
-    }
-    auto t_end = std::chrono::duration_cast<std::chrono::milliseconds>(
-            std::chrono::steady_clock::now() - t_start).count();
-    std::cout << "\n Time to run single thread with system buffer: " << t_end << "ms\n";
+  ASSERT_TRUE(initialized);
 
-    for (size_t i = 0; i < pix_buffer.size(); i++) {
-        ASSERT_EQ(pix_buffer[i], pixels[i]);
-        pix_buffer[i] = 0;
-    }
-    // --------------------------------------------------------------------------------------------
-    reader.init(file_par, false, false, 1024);
-
-    t_start = std::chrono::steady_clock::now();
-    start_buf_pos = 0;
-    for (size_t i = 0; i < this->num_bin_in_file; i++) {
-        reader.get_pix_for_bin(i, pPix_info, start_buf_pos, pix_start_num, num_bin_pix, false);
-        start_buf_pos += num_bin_pix;
-    }
-    t_end = std::chrono::duration_cast<std::chrono::milliseconds>(
-            std::chrono::steady_clock::now() - t_start).count();
-    std::cout << "\n Time to run single thread with 1024 words buffer: " << t_end << "ms\n";
+  size_t pix_start_num, num_bin_pix, start_buf_pos(0);
+  std::vector<float> pix_buffer(9 * 1000);
```

Comment:
is this 9 `NUM_PIXBLOCK_COLS`?

same question for the for loops (` for (std::size_t i = 0; i < num_bin_pix * 9; i++) { ... } `) and pixel array access (`pixels[pix_start_num * 9 + i]`)